### ISSUE TITLE
MD051 support for Gitiles section headings

### ISSUE THREAD

**Issue by dholm** [DavidAnson/markdownlint#570]
Would it be possible to add a configuration option for MD051 to support section anchors as [defined by Gitiles markdown](https://gerrit.googlesource.com/gitiles/+/HEAD/Documentation/markdown.md#named-anchors)?  I believe they pretty much follow the same convention as GitHub but they keep the case of each character rather than forcing them to lowercase.

It would also be nice if their named anchors could be supported, e.g.:

``` text
## Some Section {#name}

...

## Other Section

See [some section](#name)
```

**Comment by DavidAnson** [comment]
As I understand the specification you link to, there are a few differences:

- Translation of header text is different, notably for accented characters.
- Nested headings generate unique identifiers based on structure
- Curly bracket syntax can appear anywhere in the file

None of this is unreasonable, but overall it represents more than a trivial tweak to the logic of the existing rule.

**Comment by fekete-robert** [comment]
+1 for adding support to custom header ids. They are supported in a number of static site generators (for example, Hugo and [docusaurus](https://docusaurus.io/docs/markdown-features/toc#heading-ids)), and in these contexts this markdownlint rule throws a  false positive.

**Comment by scottamain** [comment]
I'm very interested in just the second part of this request, so that MD051 will not flag this as an invalid link:

```
## Some Section {#name}

See [some section](#name)
```

Currently, I need to disable MD051 (and use a separate linkchecker) because this use-case is important to me.

Should this be filed as a separate enhancement request?

**Comment by scottamain** [comment]
Done: https://github.com/DavidAnson/markdownlint/issues/830

**Comment by Svintooo** [comment]
+1 for adding support for gitiles flavored anchors.
My company uses gerrit with gitiles for all development, and uses markdownlint in all CI pipelines. And broken links slips through from time to time because MD051 needs to be disabled.
